acados/sim/sim_gnsf.h

```diff
@@ -0,0 +1,253 @@
+/*
+ *    This file is part of acados.
+ *
+ *    acados is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation; either
+ *    version 3 of the License, or (at your option) any later version.
+ *
+ *    acados is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ *
+ *    You should have received a copy of the GNU Lesser General Public
+ *    License along with acados; if not, write to the Free Software Foundation,
+ *    Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ * 
+ *    Author: Jonathan Frey
+ */
+
+#ifndef ACADOS_SIM_SIM_GNSF_H_
+#define ACADOS_SIM_SIM_GNSF_H_
+
+#include <stdbool.h>
+
+#include "acados/utils/timing.h"
+#include "acados/utils/types.h"
+
+#include "acados/sim/sim_common.h"
+
+#include "blasfeo/include/blasfeo_target.h"
+#include "blasfeo/include/blasfeo_common.h"
+#include "blasfeo/include/blasfeo_d_aux_ext_dep.h"
+#include "blasfeo/include/blasfeo_d_blas.h"
+#include "blasfeo/include/blasfeo_d_kernel.h"
+#include "blasfeo/include/blasfeo_i_aux_ext_dep.h"
+#include "blasfeo/include/blasfeo_d_aux.h"
+
+typedef struct
+{
+    int nx;
+    int nu;
+    int num_stages;
+    int nz;
+    int nx1;
+    int nx2;
+    int num_steps;
+    int n_out;
+    int ny;
+    int nuhat;
+} sim_gnsf_dims;
+
+
+typedef struct {
+    double *ff;
+    double *x0_1;
+    double *u_0;
+} gnsf_res_in;
+
+
+typedef struct
+{
+    // external functions
+    // phi: nonlinearity function
+    external_function_generic *phi_fun;
+    external_function_generic *phi_fun_jac_y;
+    external_function_generic *phi_jac_y_uhat;
+    // f_lo: linear output function
+    external_function_generic *f_lo_fun_jac_x1_x1dot_u_z;
+
+    // precomputed matrices
```

Comment:
Precomputed data should not belong to the model, but to the memory of the solver/integrator. 
@dkouzoup do you agree?